### PR TITLE
Update status page

### PR DESCRIPTION
**Pull request by svrnm**
This PR updates the `/status` page: We are now able to show all the per-language status. I also think spec and collector should be separated. I picked the wording from the collector page.

Preview: https://deploy-preview-3776--opentelemetry.netlify.app/status/

### CODE REVIEW

File: content/en/status.md [hunk 1]
```diff
@@ -9,42 +9,38 @@ description: Maturity-level of the main OpenTelemetry components
 
 ## {{% param title %}}
 
-OpenTelemetry is made up of [several components][main-comp], some
-language-specific and others language-agnostic. When looking for a status, make
-sure to look for the status from the right component page. For example, the
+OpenTelemetry is made up of [several components](/docs/concepts/components/),
+some language-specific and others language-agnostic. When looking for a status,
```
Comment (theletterf) [inline content/en/status.md:13]: Could we add something here about what "Status" is meant to mean or represent? Or link to the relevant verbiage.

**Review by theletterf — APPROVED** [review]

File: content/en/status.md (hunk 1 above)
Comment (svrnm) [reply content/en/status.md:13]:
```suggestion
some language-specific and others language-agnostic. When looking for a [status](/docs/specs/otel/versioning-and-stability/),
```

like this?

Comment (theletterf) [reply content/en/status.md:13]: That'll do!

**Review by cartermp — APPROVED** [review]

**Pull request merged**